tools/bazel/compile_commands/compile_commands.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python3
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+"""
+Run bazel aquery to produce compilation database (compile_commands.json)
+for particular bazel build command
+"""
+
+from __future__ import print_function
+import argparse
+import json
+import logging
+import os
+import re
+import shlex
+import subprocess
+
+
+def parse_args():
+    """
+    Parse command line arguments or show help.
+    """
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                     description=__doc__)
+    parser.add_argument("-o", "--output",
+                        default="compile_commands.json",
+                        help="output compile_commands.json file")
+    parser.add_argument("-b", "--build",
+                        help="bazel build command arguments")
+    parser.add_argument("-v", "--verbosity",
+                        default=0,
+                        action="count",
+                        help="increase output verbosity (e.g., -v or -vv)")
+    parser.add_argument("--log-format",
+                        default="[bazel] %(levelname)5s: %(message)s",
+                        help=argparse.SUPPRESS)
+
+    options = parser.parse_args()
+
+    if options.verbosity >= 2:
+        log_level = logging.DEBUG
+    elif options.verbosity >= 1:
+        log_level = logging.INFO
+    else:
+        log_level = logging.WARN
+    logging.basicConfig(level=log_level, format=options.log_format)
+
+    return options
+
+
+def split_to_list(arguments):
+    """
+    Split argument string to list
+    """
+    if type(arguments) is list:
+        return arguments
+    else:
+        return shlex.split(arguments)
```

Comment:
There is no need for this function. Use `shlex.split` where it is necessary.

Reply:
Yes, I agree that we can use `shlex.split` however `split_to_list` is more convenient and safer.
I would keep it if you dont mind.